db/db_impl_open.cc

```diff
@@ -53,12 +53,6 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
     result.write_buffer_manager.reset(
         new WriteBufferManager(result.db_write_buffer_size));
   }
-  if (result.base_background_compactions == -1) {
-    result.base_background_compactions = result.max_background_compactions;
-  }
-  if (result.base_background_compactions > result.max_background_compactions) {
-    result.base_background_compactions = result.max_background_compactions;
-  }
   result.env->IncBackgroundThreadsIfNeeded(src.max_background_compactions,
                                            Env::Priority::LOW);
   result.env->IncBackgroundThreadsIfNeeded(src.max_background_flushes,
```

Comment:
Good point. I can use GetBGJobLimits() to get values to pass to IncBackgroundThreadsIfNeeded(). It's a bit ugly, though, since this is needed in SanitizeOptions(), which is static and the DBImpl's options instance variables aren't initialized yet.